test/array.jl

```diff
@@ -123,11 +123,11 @@ end
     @test dims(da_float) == dims(da)
     @test refdims(da_float) == refdims(da)
 
-    # TODO: Yeah, what should this be? Some dimensions (i.e. where values are not explicitly enumerated) could be resizable?
-    # da_size_float = similar(da, Float64, (10, 10))
-    # @test eltype(da_size_float) == Float64
-    # @test size(da_size_float) == (10, 10)
+    da_size_float = similar(da, Float64, (10, 10))
+    @test eltype(da_size_float) == Float64
+    @test size(da_size_float) == (10, 10)
```

Comment:
@rafaqz This is failing at the moment since it's just trying to use the previous dims, but the new array is a different shape.

Does it make sense to just drop some info? I'm not sure I can think of an example of when I wouldn't want this to error.

Reply:
Hmm yeah. It's good that is caught, don't think it should error... We are going to have to think about what to do in these cases. Normally `similar` provides fallbacks instead? 

That could mean anything from `Array` to a `DimensionalArray` with `EmptyDim` dims to the same dim types without an index is a possibility.

Reply:
I would lean towards the latter. Then the only thing that breaks are `Selector`s

Reply:
Both `OffsetArrays` and `AxisArrays` fallback to the underlying array.

I think this seems like a good option, since it might help increase compatibility. I don't think it's likely a person would intentionally call this, but it might help with compatibility for a method being called.

Also, I don't think `EmptyDim` could do this quite yet.

Reply:
Ok lets go with `Array` for now and rethink it later. At least that will avoid the error.